amd_openvx/openvx/api/vx_api.cpp

```diff
@@ -9836,6 +9836,26 @@ VX_API_ENTRY vx_status VX_API_CALL vxQueryTensor(vx_tensor tensor, vx_enum attri
 					}
 				}
 #endif
+				else if (tensor_data_type == VX_TYPE_UINT32)
```

Comment:
If we are just returning the pointer for host memory, we can return it as (vx_uint8 *) for all the data types, so in the api there is no need for checking various conditions. The calling routine can typecast the pointer based on the data type. Also the (size == sizeof(vx_unit8 *)) is correct. Size is size of data pointer, not data_type